main/src/ca/uwaterloo/flix/language/phase/OccurrenceAnalyzer.scala

```diff
@@ -103,89 +123,107 @@ object OccurrenceAnalyzer {
 
-    case Expression.Float32(lit, loc) => (OccurrenceAst.Expression.Float32(lit, loc), Map.empty)
+    case Expression.Float32(lit, loc) => (OccurrenceAst.Expression.Float32(lit, loc), baseOccur)
 
-    case Expression.Float64(lit, loc) => (OccurrenceAst.Expression.Float64(lit, loc), Map.empty)
+    case Expression.Float64(lit, loc) => (OccurrenceAst.Expression.Float64(lit, loc), baseOccur)
 
-    case Expression.Int8(lit, loc) => (OccurrenceAst.Expression.Int8(lit, loc), Map.empty)
+    case Expression.Int8(lit, loc) => (OccurrenceAst.Expression.Int8(lit, loc), baseOccur)
 
-    case Expression.Int16(lit, loc) => (OccurrenceAst.Expression.Int16(lit, loc), Map.empty)
+    case Expression.Int16(lit, loc) => (OccurrenceAst.Expression.Int16(lit, loc), baseOccur)
 
-    case Expression.Int32(lit, loc) => (OccurrenceAst.Expression.Int32(lit, loc), Map.empty)
+    case Expression.Int32(lit, loc) => (OccurrenceAst.Expression.Int32(lit, loc), baseOccur)
 
-    case Expression.Int64(lit, loc) => (OccurrenceAst.Expression.Int64(lit, loc), Map.empty)
+    case Expression.Int64(lit, loc) => (OccurrenceAst.Expression.Int64(lit, loc), baseOccur)
 
-    case Expression.BigInt(lit, loc) => (OccurrenceAst.Expression.BigInt(lit, loc), Map.empty)
+    case Expression.BigInt(lit, loc) => (OccurrenceAst.Expression.BigInt(lit, loc), baseOccur)
 
-    case Expression.Str(lit, loc) => (OccurrenceAst.Expression.Str(lit, loc), Map.empty)
+    case Expression.Str(lit, loc) => (OccurrenceAst.Expression.Str(lit, loc), baseOccur)
 
-    case Expression.Var(sym, tpe, loc) => (OccurrenceAst.Expression.Var(sym, tpe, loc), Map(sym -> Once))
+    case Expression.Var(sym, tpe, loc) => (OccurrenceAst.Expression.Var(sym, tpe, loc), OccurInfo(Map(sym -> Once), Map.empty, 1))
 
     case Expression.Closure(sym, freeVars, tpe, loc) =>
       val (fv, o) = freeVars.foldRight[(List[OccurrenceAst.FreeVar], Map[Symbol.VarSym, Occur])]((List.empty, Map.empty)) {
         case (LiftedAst.FreeVar(sym, tpe), (fvs, o)) =>
           (OccurrenceAst.FreeVar(sym, tpe) :: fvs, o + (sym -> DontInline))
       }
-      (OccurrenceAst.Expression.Closure(sym, fv, tpe, loc), o)
+      (OccurrenceAst.Expression.Closure(sym, fv, tpe, loc), OccurInfo(o, Map.empty, 1+fv.length))
 
     case Expression.ApplyClo(exp, args, tpe, purity, loc) =>
       val (e, o1) = visitExp(exp)
       val (as, o2) = visitExps(args)
       val o3 = combineAllSeq(o1, o2)
-      (OccurrenceAst.Expression.ApplyClo(e, as, tpe, purity, loc), o3)
+      exp match {
+        case Expression.Closure(sym, _, _, _) =>
+          val o4 = OccurInfo(Map.empty, Map(sym -> Once), 0)
+          val o5 = combineAllSeq(o3, o4)
+          (OccurrenceAst.Expression.ApplyClo(e, as, tpe, purity, loc), o5.copy(codeSize = o5.codeSize+1))
+        case _ => (OccurrenceAst.Expression.ApplyClo(e, as, tpe, purity, loc), o3.copy(codeSize = o3.codeSize+1))
+      }
```

Comment:
@magnus-madsen we are a bit unsure what to do here. We can see that we are often hitting the wild card case which means we don't know which closure is being called. But we still want to collect occurrence information for the apply closure. Do you have any idea on what would be a correct solution?